src/Transports.AspNetCore/HttpGetValidationRule.cs

```diff
@@ -0,0 +1,23 @@
+#nullable enable
+
+using GraphQL.Server.Transports.AspNetCore.Errors;
+using GraphQL.Validation;
+using GraphQLParser.AST;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Validates that HTTP GET requests only execute queries; not mutations or subscriptions.
+/// </summary>
+public sealed class HttpGetValidationRule : IValidationRule
+{
+    /// <inheritdoc/>
+    public ValueTask<INodeVisitor?> ValidateAsync(ValidationContext context)
+    {
+        if (context.Operation.Operation != OperationType.Query)
+        {
+            context.ReportError(new HttpMethodValidationError(context.Document.Source, context.Operation, "Only query operations allowed for GET requests."));
```

Comment:
This message uses wording _allowed_, other one from `HttpPostValidationRule` uses _supported_. I suppose it's so for first to conform http-spec:
> GET requests MUST NOT be used for executing mutation operations. If the values of {query} and {operationName} indicate that a mutation operation is to be executed, the server SHOULD immediately respond with error status code 405 (Method Not Allowed) and halt execution. 

But what is about subscriptions? I see nothing in http-spec about subscriptions.

Reply:
Also note - 405 http code, not 400.

Reply:
Subscriptions over HTTP are only possible over WebSocket connections.

Reply:
Yes I am familiar with the spec. Ours will return 400 because of the design. As the spec says “should” and not “must” it is still allowed. I actually thought we already discussed this but perhaps not.

Reply:
And further - I doubt `HttpGetValidationRule` should be a validation rule at all. See _immediately respond_ wording above.

Reply:
I agree but the parsing has not occurred prior. You are welcome to create a PR that works better. For now this is an improvement over nothing, which is what happens now.

Reply:
#794

Reply:
Option - add some sort of 
```cs
public interface IHttpRequestValidationError
{
    HttpStatusCode ResponseCode { get; }
}
```
then
```cs
public class HttpMethodValidationError : ValidationError, IHttpRequestValidationError
{
    public HttpStatusCode ResponseCode => HttpStatusCode.MethodNotAllowed;
}
```
then
```cs
 var statusCode = Options.ValidationErrorsReturnBadRequest && !result.Executed
            ? result.Errors.OfType<IHttpRequestValidationError>().FirstOrDefault()?.ResponseCode ?? HttpStatusCode.BadRequest
            : HttpStatusCode.OK;
```

Reply:
It does not allow to immediately respond but allows to return proper http code.